File: Arrays/3541. Find Most Frequent Vowel and Consonant/3541. Find Most Frequent Vowel and Consonant.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels = [0] * 26
        consonants = [0] * 26

        for char in s:
            i = ord(char) - ord("a")
            if char in "aeiou":
                vowels[i] += 1
            else:
                consonants[i] += 1
        
        return max(vowels) + max(consonants)

# Time: O(n)
# Space: O(1)

File: Arrays/3541. Find Most Frequent Vowel and Consonant/3541. Find Most Frequent Vowel and Consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function (s) {
  const vowels = new Array(26).fill(0);
  const consonants = new Array(26).fill(0);

  for (let char of s) {
    const i = char.charCodeAt(0) - "a".charCodeAt(0);
    if ("aeiou".includes(char)) vowels[i]++;
    else consonants[i]++;
  }

  return Math.max(...vowels) + Math.max(...consonants);
};

// Time: O(n)
// Space: O(1)
